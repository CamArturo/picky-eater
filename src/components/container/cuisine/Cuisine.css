.cuisine-header {
  background: url("../../../assets/food-200.jpg");
  height: 200px; }
  .cuisine-header h2 {
    font-family: Signika Negative, serif;
    font-style: normal;
    font-weight: bold;
    line-height: 56px;
    font-size: 36px;
    letter-spacing: 0.04em;
    text-align: center;
    padding: 48px 0 0 0;
    margin: 0;
    color: #fff;
    text-shadow: 3px 3px 4px rgba(0, 0, 0, 0.7); }

.cuisine-container {
  display: flex;
  flex-wrap: wrap;
  height: 100vh; }
  .cuisine-container .return-restaurants-container {
    width: 100%;
    height: 100px;
    margin: 1rem 0;
    display: flex;
    justify-content: center;
    align-items: center; }
    .cuisine-container .return-restaurants-container .return-restaurants {
      width: auto;
      height: 40px;
      font-family: Catamaran, serif;
      font-weight: bold;
      font-size: 18px;
      line-height: 37px;
      letter-spacing: 2px;
      color: #fff;
      background-color: #0956B2;
      border: none;
      margin: 0; }
      .cuisine-container .return-restaurants-container .return-restaurants:hover {
        background-color: #1468CC;
        color: #fff;
        cursor: pointer; }
  .cuisine-container .cuisine-btn {
    width: 10%;
    min-height: 100px;
    height: 100px;
    display: flex;
    justify-content: center;
    align-items: center;
    border: 1px solid #cecece; }
    .cuisine-container .cuisine-btn:hover {
      background-color: #cecece; }
    .cuisine-container .cuisine-btn h3 {
      pointer-events: none;
      font-family: Catamaran, serif;
      font-style: italic;
      font-weight: normal;
      line-height: 150%;
      letter-spacing: 0.04em;
      color: #000; }

.loading-gif {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  width: 100%; }

.no-locations {
  height: 100vh;
  margin: 1rem auto 0; }

/* Large Devices, Wide Screens */
/* Medium Devices, Desktops */
/* Small Devices, Tablets */
/* Extra Small Devices, Phones */
/* Custom, iPhone Retina */

/*# sourceMappingURL=Cuisine.css.map */
