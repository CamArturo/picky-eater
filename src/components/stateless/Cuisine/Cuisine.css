.cuisine-container-header {
  background: url("../../../assets/food-200.jpg");
  height: 200px; }
  .cuisine-container-header h2 {
    font-family: Signika Negative, serif;
    font-style: normal;
    font-weight: bold;
    line-height: 56px;
    font-size: 36px;
    letter-spacing: 0.04em;
    text-align: center;
    padding: 48px 0 0 0;
    margin: 0;
    color: #fff;
    text-shadow: 3px 3px 4px rgba(0, 0, 0, 0.7); }

.cuisine-container {
  display: flex;
  flex-wrap: wrap; }
  .cuisine-container .cuisine-card {
    width: 240px;
    height: 200px;
    display: flex;
    justify-content: center;
    align-items: center; }

/* Large Devices, Wide Screens */
/* Medium Devices, Desktops */
/* Small Devices, Tablets */
/* Extra Small Devices, Phones */
/* Custom, iPhone Retina */

/*# sourceMappingURL=Cuisine.css.map */
