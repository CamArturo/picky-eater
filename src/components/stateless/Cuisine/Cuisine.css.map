{
"version": 3,
"mappings": "AAAA,eAAgB;EACd,UAAU,EAAE,mCAAmC;EAC/C,MAAM,EAAE,KAAK;EACb,kBAAG;IACD,WAAW,EAAE,uBAAuB;IACpC,UAAU,EAAE,MAAM;IAClB,WAAW,EAAE,IAAI;IACjB,WAAW,EAAE,IAAI;IACjB,SAAS,EAAE,IAAI;IACf,cAAc,EAAE,MAAM;IACtB,UAAU,EAAE,MAAM;IAClB,OAAO,EAAE,UAAU;IACnB,MAAM,EAAE,CAAC;IAET,KAAK,EAAE,IAAI;IACX,WAAW,EAAE,8BAA8B;;AAI/C,kBAAmB;EACjB,OAAO,EAAE,IAAI;EACb,SAAS,EAAE,IAAI;EACf,gCAAc;IACZ,KAAK,EAAE,KAAK;IACZ,MAAM,EAAE,KAAK;IACb,OAAO,EAAE,IAAI;IACb,eAAe,EAAE,MAAM;IACvB,WAAW,EAAE,MAAM;IACnB,MAAM,EAAE,iBAAiB;IACzB,mCAAG;MACD,KAAK,EAAE,KAAK;MACZ,MAAM,EAAE,IAAI;MAEZ,WAAW,EAAE,gBAAgB;MAC7B,UAAU,EAAE,MAAM;MAClB,WAAW,EAAE,MAAM;MACnB,WAAW,EAAE,IAAI;MACjB,SAAS,EAAE,IAAI;MACf,cAAc,EAAE,MAAM;MAEtB,KAAK,EAAE,IAAI;;AAKjB,iCAAiC;AAKjC,8BAA8B;AAK9B,4BAA4B;AAK5B,iCAAiC;AAKjC,2BAA2B",
"sources": ["Cuisine.scss"],
"names": [],
"file": "Cuisine.css"
}