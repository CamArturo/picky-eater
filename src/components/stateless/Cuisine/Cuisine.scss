.cuisine-header {
  background: url("../../../assets/food-200.jpg");
  height: 200px;
  h2 {
    font-family: Signika Negative, serif;
    font-style: normal;
    font-weight: bold;
    line-height: 56px;
    font-size: 36px;
    letter-spacing: 0.04em;
    text-align: center;
    padding: 48px 0 0 0;
    margin: 0;

    color: #fff;
    text-shadow: 3px 3px 4px rgba(0, 0, 0, 0.7);
  }
}

.cuisine-container {
  display: flex;
  flex-wrap: wrap;
  .cuisine-card {
    width: 240px;
    height: 200px;
    display: flex;
    justify-content: center;
    align-items: center;
    border: 1px solid #cecece;
    h3 {
      width: 124px;
      height: 51px;

      font-family: Catamaran, serif;
      font-style: italic;
      font-weight: normal;
      line-height: 150%;
      font-size: 2rem;
      letter-spacing: 0.04em;

      color: #000;
    }
  }
}

/* Large Devices, Wide Screens */
@media only screen and (max-width: 1200px) {

}

/* Medium Devices, Desktops */
@media only screen and (max-width: 992px) {

}

/* Small Devices, Tablets */
@media only screen and (max-width: 768px) {

}

/* Extra Small Devices, Phones */
@media only screen and (max-width: 480px) {

}

/* Custom, iPhone Retina */
@media only screen and (max-width: 320px) {

}