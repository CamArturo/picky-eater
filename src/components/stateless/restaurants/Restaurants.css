.restaurant-header {
  background: url("../../../assets/restaurants.jpg");
  height: 200px; }
  .restaurant-header h2 {
    font-family: Signika Negative, serif;
    font-style: normal;
    font-weight: bold;
    line-height: 56px;
    font-size: 36px;
    letter-spacing: 0.04em;
    text-align: center;
    padding: 48px 0 0 0;
    margin: 0;
    color: #fff;
    text-shadow: 3px 3px 4px rgba(0, 0, 0, 0.7); }

.restaurant-container {
  min-height: 100vh;
  display: flex;
  flex-wrap: wrap; }
  .restaurant-container article {
    width: 20%;
    height: 100px;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 1rem;
    text-align: center;
    background-color: #c4c4c4; }

/*# sourceMappingURL=Restaurants.css.map */
