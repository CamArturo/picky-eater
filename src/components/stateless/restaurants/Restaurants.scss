.restaurant-container {
  min-height: 100vh;
}