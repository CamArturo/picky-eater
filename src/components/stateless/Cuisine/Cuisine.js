import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Loading from '../Loading/Loading';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { getCuisines } from '../../../api/Api';
import { sendCuisinesToStore } from '../../../actions/cuisines';
import './Cuisine.css';

class Cuisine extends Component {
  constructor (props) {
    super(props);
  }

  componentDidMount () {
    if (this.props.chosenCityID) {
      this.storeCuisines(this.props.chosenCityID);
    }
  }

  displayNoLocations = () => {
    return (
      <section className="no-locations">
        <p>No Locations Chosen</p>
      </section>
    );
  };

  storeCuisines = async (chosenCityID) => {
    const cuisines = await getCuisines(chosenCityID);
    this.props.loadCuisines(cuisines.cuisines);
  };

  displayCuisines = () => {
    return this.props.availableCuisines.map((cuisine, index) => (
      <button className="cuisine-btn" key={`key${index}`}>
        <h3>{cuisine.cuisine.cuisine_name}</h3>
      </button>
    ));
  };

  render () {
    return (
      <div>
        <section className="cuisine-header">
          <h2>What do you NOT want to eat?</h2>
        </section>
        <section className="cuisine-container">
          {
            this.props.chosenCity &&
            this.displayNoLocations()
          }
          {
            this.props.availableCuisines.length > 0 ?
              this.displayCuisines() : <Loading/>
          }
        </section>
      </div>
    );
  }
}

// Cuisine.propTypes = {};

const mapStateToProps = (state) => ({
  availableCuisines: state.availableCuisines,
  chosenCityID: state.chosenCity.id
});

const mapDispatchToProps = (dispatch) => ({
  loadCuisines: (cities) => dispatch(sendCuisinesToStore(cities))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Cuisine));
