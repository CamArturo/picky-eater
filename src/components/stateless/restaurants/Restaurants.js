import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import './Restaurants.css'

export class Restaurants extends Component {
  constructor (props) {
    super(props);

  }

  displayRestaurants = () => {
    return this.props.restaurants.map((restaurant, index) => (
      <article key={`key${index}`}>
        <h2>{restaurant}</h2>
      </article>
    ));
  };

  render () {
    return (
      <div>
        <section className="restaurant-header">
          <h2>These are you results</h2>
        </section>
        <section className="restaurant-container">
          {
            this.props.restaurants.length > 0 &&
            this.displayRestaurants()
          }
        </section>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  restaurants: state.restaurants
});

Restaurants.propTypes = {
  restaurants: PropTypes.array.isRequired
};

export default connect(mapStateToProps, null)(Restaurants);
