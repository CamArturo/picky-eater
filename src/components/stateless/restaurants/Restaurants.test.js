import React from 'react';
import { Restaurants, mapStateToProps } from './Restaurants';
import { shallow } from 'enzyme';
import { mockListRestaurants } from '../../../mockData';
import * as mock from '../../../mockData';
import { mapDispatchToProps } from '../../container/cuisine/Cuisine';

describe('Restaurants page tests', () => {

  let wrapper;
  let mockState;

  beforeEach(() => {
    mockState = {
      restaurants: mockListRestaurants,
    };

    wrapper = shallow(
      <Restaurants
        restaurants={mockListRestaurants}
      />);

  });

  it('Should match the snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });

  it('should mapStateToProps', () => {
    const actual = mapStateToProps(mockState);
    const expected = {
      restaurants: mockListRestaurants
    };
    expect(actual).toEqual(expected);
  });
});