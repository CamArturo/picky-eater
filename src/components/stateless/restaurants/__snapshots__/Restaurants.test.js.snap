// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restaurants page tests Should match the snapshot 1`] = `undefined`;
